helper/common/shared_state.go

```diff
@@ -9,23 +9,23 @@ import (
 
 // Used to set variables which we need to access later in the build, where
 // state bag and config information won't work
-func sharedStateFilename(suffix string) string {
+func sharedStateFilename(suffix string, buildName string) string {
 	uuid := os.Getenv("PACKER_RUN_UUID")
-	return filepath.Join(os.TempDir(), fmt.Sprintf("packer-%s-%s", uuid, suffix))
+	return filepath.Join(os.TempDir(), fmt.Sprintf("packer-%s-%s-%s", uuid, suffix, buildName))
 }
 
-func SetSharedState(key string, value string) error {
-	return ioutil.WriteFile(sharedStateFilename(key), []byte(value), 0600)
+func SetSharedState(key string, value string, buildName string) error {
```

Comment:
instead of changing the function signature, can we just modify the key before calling this?

Reply:
We could, but that'd require future users of the function to know they need to have the build name in the signature.

Reply:
I'd rather have it explicit, even though I have to change the function signature.